kernel/mempool.c

```diff
@@ -13,6 +13,14 @@
 #include <sys/math_extras.h>
 #include <stdbool.h>
 
+#ifdef CONFIG_SYS_TIMEOUT_LEGACY_API
```

Comment:
Could we have global K_TICKS_FOREVER, K_TICKS_NO_WAIT instead of local definitions?